### DISCUSSION TITLE
Pollute Tracking UTMs and Other Parameters

### DISCUSSION THREAD

**Discussion by alicezwatson** [dhowe/AdNauseam#2479] [Ideas]
I would love it if AdNauseam had an option for replacing UTM (and other non-site-breaking) parameters with plausible lies. If enough people used it, we could pollute marketing data in a way that makes it expensive to clean and useless in general.

For instance, http://example.com/?utm_source=facebook&utm_medium=social&utm_campaign=privacy_invasion&utm_term=kittens&utm_content=pfk0wqSZgVwlkh0akzSY&search_term=cute%20kitten%20pictures

could be replaced with http://example.com/?utm_source=myspace&utm_medium=other&utm_campaign=fall_ad_campaign&utm_term=forks&utm_content=BQcF8qtgz11sANoAAl1b&search_term=cute%20kitten%20pictures

Each of the standard UTM parameters is replaced with something that will cause issues for analysis and marketing spend tracking, while leaving something like `search_term` intact.

The UI for it could show URL parameters and let the user check a box to scramble them or leave them untouched.

**Comment by pohutukawa** [comment]
Similarly it might be a grand idea to also allow for spoofing the new YouTube link parameter with bogus info.

**Comment by dhowe** [comment]
this is an interesting idea... I wonder a) what would be needed in terms of UI (perhaps just a toggle ?) and b) the implementation time/difficulty?  @mneunomne ?

**Comment by alicezwatson** [comment]
I started taking a stab at making my own Firefox extension to do this, but I've never made one before, so it's barely working. However, the code and concept are there. So if it helps with developing this feature for AdNauseam, by all means have at it. I think it would be more effective as part of this established project than a separate add-on anyway.

https://codeberg.org/alicewatson/paraminator
